forge-core/src/main/java/forge/util/FileSection.java

```diff
@@ -105,25 +105,25 @@ public static Map<String, List<String>> parseSections(final List<String> lines)
         int lineNumber = 0;
         String section = null;
 
-        do{
-            String line = lines.get(lineNumber++);
-            if (line.startsWith("[") && line.endsWith("]")) {
-                section = line.substring(1, line.length() - 1);
-                if(!result.containsKey(section)) {
-                    result.put(section, new ArrayList<>());
+        if(null != lines && !lines.isEmpty()){
+            do{
```

Comment:
Why is this a do/while instead of a for loop? 

for(int lineNumber = 0; lineNumber < lines.size(); lineNumber++) {}

Reply:
it is code optimization. it allows to do the same thing with less line of code and less if statetemens inside the loop.

Reply:
I worry that you embed the increment here. `lineNumber++` which might confuse people of what is actually going on, and if someone decides to add code that interacts with lineNumber after this increment, it won't be the same value as what gets used on the line itself. I wouldn't call that more optimal, as I'm sure the compiler handles the scenarios pretty similarly.

    ```java
    if(null == lines || lines.isEmpty()) {
        return result;
    }
    for(int lineNumber = 0; lineNumber<lines.size(); lineNumber++) {
                String line = lines.get(lineNumber);
                if (line.startsWith("[") && line.endsWith("]")) {
                    section = line.substring(1, line.length() - 1);
                    if(!result.containsKey(section)) {
                        result.put(section, new ArrayList<>());
                    }
                }
                else if(null != section && !line.isEmpty() && !line.startsWith("#")){
                    result.get(section).add(line);
                }
            }
``` 

In fact, we don't even care about the line number, so we could just do...

```java
if (lines == null) {
   return result;
}

for(String line : lines) {
   if (line.startsWith("#") continue;

   if (line.startsWith("[") && line.endsWith("]")) {
          section = line.substring(1, line.length() - 1);
          if(!result.containsKey(section)) {
                 result.put(section, new ArrayList<>());
          }
    } else if(null != section && !line.isEmpty()){
        result.get(section).add(line);
    }
}
```

Reply:
This also seems like we're potentially adding lines to sections that can be just spaces now? since we aren't trimming or string comparison equivilants of that.

Reply:
Hello @tehdiplomat , Indeed, I agree the for loop is more friendly. The algorithm do not require a do/while at all, didn't realize that. I have updated the code with your sugestion and have added a trim to the line as well.